### PR TITLE
Blog post of Mutiny retry facilities

### PR DESCRIPTION
**Pull request by cescoffier**
CC @jponge

### CODE REVIEW

**Comment by jponge** [comment]
Did you forget the text file?

**Comment by cescoffier** [comment]
of course I did.Should be better now.

File: _posts/2020-10-13-uni-retry.adoc [hunk 1]
```diff
@@ -0,0 +1,170 @@
+---
+layout: post
+title: Mutiny - How does retry retry?
```
Comment (jponge) [inline _posts/2020-10-13-uni-retry.adoc:3]: You should create more drama: `How does retry... retries?`

File: _posts/2020-10-13-uni-retry.adoc [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+image:/assets/images/posts/mutiny-retry/retry.png[]
+
+And this until it gets a successful response.
+In other words, retrying is resubscribing.
+
+## How many times should I retry?
+
+That’s always a good question. 
+Should I retry _indefinitely_?
+Most probably, no.
```
Comment (jponge) [inline _posts/2020-10-13-uni-retry.adoc:97]: s/no/not

File: _posts/2020-10-13-uni-retry.adoc (hunk 1 above)
Comment (cescoffier) [reply _posts/2020-10-13-uni-retry.adoc]: Good point! Fixed!

File: _posts/2020-10-13-uni-retry.adoc (hunk 2 above)
Comment (cescoffier) [reply _posts/2020-10-13-uni-retry.adoc]: Fixed

File: _posts/2020-10-13-uni-retry.adoc [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+[source, java]
+----
+Uni<String> uni = invokePickyService(client);
+uni.subscribe().with(
+    resp -> System.out.println("Success: " + resp),
+    failure -> System.out.println("Failed: " + failure.getMessage())
+);
+----
+
+In Quarkus,  most of the time, you return the `Uni`, and Quarkus subscribes to it.
```
Comment (kenfinnigan) [inline _posts/2020-10-13-uni-retry.adoc:55]: Looks like a doubled space after the first comma?

File: _posts/2020-10-13-uni-retry.adoc [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+In Quarkus,  most of the time, you return the `Uni`, and Quarkus subscribes to it. 
+So, don't be mistaken, there is a subscription, you may not see it.
+
+This laziness is the retry secret.
+In the following sequence diagram, you can see that the request is sent when we get a subscriber:
+
+image:/assets/images/posts/mutiny-retry/subscription.png[]
+
+
+An interesting consequence of this is that if you subscribe twice, you emit two requests, and so gets two responses:
```
Comment (kenfinnigan) [inline _posts/2020-10-13-uni-retry.adoc:64]: s/gets/get

File: _posts/2020-10-13-uni-retry.adoc [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+In this snippet, we retry _indefinitely_ until we get a successful result.
+
+But, how does it work under the hood?
+It’s quite simple.
+If it gets a failure, it just re-subscribes:
+
+image:/assets/images/posts/mutiny-retry/retry.png[]
+
+And this until it gets a successful response.
```
Comment (kenfinnigan) [inline _posts/2020-10-13-uni-retry.adoc:90]: I'm not sure what "And this" is referring to here

**Review by kenfinnigan — CHANGES_REQUESTED** [review]

File: _posts/2020-10-13-uni-retry.adoc (hunk 3 above)
Comment (cescoffier) [reply _posts/2020-10-13-uni-retry.adoc:55]: Wow... how can you see that!

**Comment by cescoffier** [comment]
@kenfinnigan, All fixed!

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**